content/developers/dogstatsd/unix_socket.md

```diff
@@ -0,0 +1,122 @@
+
+## Setup
+
+### Agent
+
+Edit your `datadog.yaml` file to set the `dogstatsd_socket` option to the path where DogStatsD should create its listening socket:
+
+```
+dogstatsd_socket: /var/run/datadog/dsd.socket
+```
+
+Then [restart your Agent][1]. You can also set the socket path via the `DD_DOGSTATSD_SOCKET` environment variable.
+
+### Client
+
+The following DogStatsD client libraries support UDS traffic:
+
+| Language | Library                            |
+| :----    | :----                              |
+| Golang   | [DataDog/datadog-go][3]            |
+| Java     | [DataDog/java-dogstatsd-client][4] |
+| Python   | [DataDog/datadogpy][5]             |
+| Ruby     | [DataDog/dogstatsd-ruby][6]        |
+
+Refer to the library's documentation on how to enable UDS traffic.
+
+**Note:** As with UDP, enabling client-side buffering is highly recommended to improve performance on heavy traffic. Refer to your client library's documentation for instructions.
+
+### Accessing the socket across containers
+
+When running in a containerized environment, the socket file needs to be accessible to the client containers. To achieve this, we recommend mounting a host directory on both sides (read-only in your client containers, read-write in the Agent container).
+
+Mounting the parent folder instead of the individual socket enables socket communication to persist across DogStatsD restarts.
+
+#### Docker: bind mount
+
+* Start the Agent container with `-v /var/run/datadog:/var/run/datadog`
+* Start your containers with `-v /var/run/datadog:/var/run/datadog:ro`
+
+#### Kubernetes: `hostPath` volume
+
+Mount the folder in your `datadog-agent` container:
+
+```
+volumeMounts:
+  - name: dsdsocket
+    mountPath: /var/run/datadog
+...
+volumes:
+- hostPath:
+    path: /var/run/datadog/
+  name: dsdsocket
+```
+
+Expose the same folder in your client containers:
+
+```
+volumeMounts:
+  - name: dsdsocket
+    mountPath: /var/run/datadog
+    readOnly: true
+...
+volumes:
+- hostPath:
+    path: /var/run/datadog/    
+  name: dsdsocket
+```
+
+## Using Origin Detection for container tagging
+
+Origin Detection allows DogStatsD to detect where the container metrics come from, and tag metrics automatically. When this mode is enabled, all metrics received via UDS will be tagged by the same container tags as Autodiscovery metrics. **Note:** `container_id`, `container_name` and `pod_name` tags will not be added, to avoid creating too many custom metric contexts.
+
+To enable this, enable the `dogstatsd_origin_detection` option in your `datadog.yaml`, or set the environment variable `DD_DOGSTATSD_ORIGIN_DETECTION=true`, and [restart your Agent][1].
+
+When running inside a container, DogStatsd needs to run in the host PID namespace for origin detection to work reliably. You can enable this via the docker `--pid=host` flag.
```

Comment:
"Origin detection" needs consistent capitalization. Either capitalize it here, or lowercase it above.